sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/PostgreSQLDialect.scala

```diff
@@ -46,4 +46,19 @@ object PostgreSQLDialect {
       }
     }
   }
+
+  object CastToTimestamp extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = {
+      val conf = SQLConf.get
+      if (conf.usePostgreSQLDialect) {
+        plan.transformExpressions {
+          case Cast(child, dataType, timeZoneId)
+            if dataType == TimestampType =>
```

Comment:
We can leave `cast timestamp to timestamp` case to Optimizer to do optimization.

Reply:
Oh, sorry. Here, I mean that we should change the if condition to: `if child.dataType != TimestampType && dataType == TimestampType && =>`. Because `Optimizer`, currently, can not optimize Pg cast.

Reply:
Got this. I will update this.